### PR TITLE
refactor: add redux-form with validator on login

### PR DESCRIPTION
**Pull request by isaadabbasi**

### CODE REVIEW

File: client/src/components/LoginForm/LoginForm.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+import React from 'react';
+import { Input, Button } from 'reactstrap';
+import './LoginForm.css';
+import { Field, reduxForm } from 'redux-form';
+import renderField from '../RenderField';
+import validate from './LoginValidator';
+
+type Props = {
+  onSubmit(e: any): void,
```
Comment (mashhoodr) [inline client/src/components/LoginForm/LoginForm.js]: @isaadabbasi dont we have an event type for above?

File: client/src/components/LoginForm/LoginForm.js [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+};
+
+let LoginForm = (props: Props) => {
+  const { valid, onSubmit } = props;
+  console.log(props);
```
Comment (mashhoodr) [inline client/src/components/LoginForm/LoginForm.js]: Remove logs

File: client/src/components/LoginForm/LoginForm.js [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+  return (
+    <form id="login-form" onSubmit={onSubmit}>
```
Comment (mashhoodr) [inline client/src/components/LoginForm/LoginForm.js]: why do we need the id?

File: client/src/components/LoginForm/LoginForm.js [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+      <Field
+        type="text"
+        placeholder="User Name"
+        className="form-control"
+        name="username"
+        component={renderField}
```
Comment (mashhoodr) [inline client/src/components/LoginForm/LoginForm.js]: renderField needs renaming

File: client/src/components/LoginForm/LoginValidator.js [hunk 5]
```diff
@@ -0,0 +1,13 @@
+import { isEmail, isLowercase, isLength } from 'validator';
```
Comment (mashhoodr) [inline client/src/components/LoginForm/LoginValidator.js]: are these being used?

File: client/src/components/LoginForm/LoginForm.js (hunk 3 above)
Comment (isaadabbasi) [reply client/src/components/LoginForm/LoginForm.js]: for css, to limit the styles in scope of this particular id, otherwise it scatters around the app.

Comment (mashhoodr) [reply client/src/components/LoginForm/LoginForm.js]: arnt styles encapsulated to the component?

**Comment by mashhoodr** [comment]
@isaadabbasi ready to merge? Didnt see the commit for renderField

**Comment by isaadabbasi** [comment]
@mashhoodr, Yes, we can merge it now. i will pluck another branch for that task as it will have 4 to 5 changes.

**Pull request merged**